osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -105,20 +105,17 @@ Status FirehoseLogForwarder::setUp() {
   }
 
   // Make sure we can connect to designated stream
-  Aws::Firehose::Model::ListDeliveryStreamsRequest r;
-  auto result = client_->ListDeliveryStreams(r).GetResult();
-  std::vector<std::string> stream_names = result.GetDeliveryStreamNames();
-
-  if (std::find(stream_names.begin(),
-                stream_names.end(),
-                FLAGS_aws_firehose_stream) == stream_names.end()) {
-    return Status(1,
-                  "Could not find Firehose delivery stream: " +
-                      FLAGS_aws_firehose_stream);
+  Aws::Firehose::Model::DescribeDeliveryStreamRequest r;
+  r.SetDeliveryStreamName(FLAGS_aws_firehose_stream);
+  auto result = client_->DescribeDeliveryStream(r).GetResult();
+  auto description = result.GetDeliveryStreamDescription();
+  auto stream_name = description.GetDeliveryStreamName();
+  if (stream_name.empty()){
```

Comment:
@hackgnar I'd be mostly concerned that, if `FLAGS_aws_firehose_stream` is not a valid stream, what does `client_->DescribeDeliveryStream(r)` return? Is it an object with some sort of internal error state? How do you know if it was successful? Are you certain it's always ok to call `GetResult()` on that object in every state that the initial function call could leave the resulting object in?

You're basically doing this:

```cpp
if (client_->DescribeDeliveryStream(r).GetResult().GetDeliveryStreamDescription().GetDeliveryStreamName().empty()) {
 // ....
}
```

Thus, if you want to safely do this, you MUST exhaustively reason about the return type of every single one of those methods, their error state, and ensure that they will _never_ return a nullptr, else you will segfault and crash the entire osqueryd process.